### DISCUSSION TITLE
High Input Lag in Osu! lazer

### DISCUSSION THREAD

**Discussion by reddude123** [ppy/osu#15940] [Q&A]
Recently, I download osu lazer and have been encountering very high input lag while playing any maps. I have turned of all effects that may affect my fps. Today, i found out that it goes away when i switch to single-thread mode instead of multi-thread mode for some reason. On osu stable i used compatibility mode because I would also get very high input lag in the game without it. Is there going to be a compatibility mode in osu lazer because I don't want to destroy my performance by switching to single-thread mode.

**Comment by integerrr** [comment]
I'm not sure I understand how single-threaded mode help with input lag and destroy the performance at the same time. Can you provide a bit more detail about your hardware and system configurations? 

Do note though, optimizations are constantly in the works so the performance can still improve in future releases.

**Reply by reddude123** [reply]
I have a Asus Tuf FX505DT laptop and its specifications are:
AMD Radeon something i forgot
Ryzen 7 3750H
1660 Ti
16 GB RAM
I also made sure to make osu lazer use my dedicated gpu instead of my integrated and there was still no change in input lag. I also made sure to play in fullscreen. Also when I use single thread mode my game stutters a lot and had a much lower frame rate

**Comment by Moltanicaa** [comment]
Try turn off High precigion mouse

**Comment by bdach** [comment]
i can't say anything for sure without a video showing what you experience. preferably with the frame graph showing (press ctrl+f11 twice from a state where there is no fps display).

**Comment by peppy** [comment]
Probably same as always: game not running in exclusive full screen. It's always the reason for input lag. Still many cases it isn't invoked when it should be. Brought this up in the past, but it'd be really cool if there was a way to detect whether a game is running exclusive or not.

**Reply by reddude123** [reply]
I have tried running the game in exclusive full screen and tried turning off high precision mouse but there has still been no change to the input lag.

**Reply by peppy** [reply]
My point is it probably isn't. No matter how much you try, if it's not running exclusive fullscreen it's not running.

**Reply by smoogipoo** [reply]
Switchable GPU laptops are notoriously hard to get exclusive fullscreen with OpenGL. Your best shot is to wait for compatibility mode or similar.

**Comment by Frous-s** [comment]
I have exactly this problem, I play with an AMD(ryzen 5 2600 and RX 580 8GB) pc and a CTL 4100WL, and I notice my cursor doesn't reach where I had aimed initially, also happens a lot more with wiggle mod.

**Comment by Toxic1594** [comment]
same here , but also for osu normal when i dont enable compatibly mode and borderless  and detect performance problems , these 2 features dont exist in osu!lazer -> unplayable , sad that they cant fix it since over 8 years

**Reply by bdach** [reply]
your comment has no actual details to work with and therefore is unactionable.
